mantidproject/mantid
# Modify composite function attributes API

**Description of work.**
This PR modifies the API of attributes in Composite function, so it is similar to the API for setting/getting parameters of a composite function. This is to avoid having to write different code for manipulating attributes on composite functions and IFunctions.
Fundamentally, the changes can be seen in the script, where all the attributes are now printed, with an appropriate prefix.
```
func = FunctionFactory.Instance().createInitialized("name=GausOsc,A=0.2,Sigma=0.2,Frequency=0.1,Phi=0;name=TeixeiraWaterSQE,Q=8.9884656743115785e+307,WorkspaceIndex=2147483647,Height=1,DiffCoeff=2.3,Tau=1.25,Centre=0")
print("Composite attributes", func.attributeNames())
```

**To test:**
1. Code review
2. Generally these changes should be covered by unittests, but testing the following fitting interfaces would be a good place to start:
3. Test the workbench fitting browser by loading data and adding an appropriate fit function - perform a fit.
4. Test the Muon analysis fitting browser: Open the Muon Analysis GUI -> Load any data -> Go to fitting -> Add and remove some fit functions -> Run the fit
5. Test indirect: Open the Indirect Data Analayis GUI -> ConvFit -> Load data -> Add fit functions, e.g 2 Lorentzians  -> Click the full function view tickbox -> Run the fit

 



*There is no associated issue.*





---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
